packages/react/src/components/ToggleGroup/ToggleGroup.tsx

```diff
@@ -0,0 +1,79 @@
+import type { HTMLAttributes } from 'react';
+import React, { createContext, forwardRef, useId, useState } from 'react';
+import cn from 'classnames';
+
+import { RovingTabindexRoot } from '../../utility-components/RovingTabIndex';
+
+import classes from './ToggleGroup.module.css';
+
+export type ToggleGroupContextProps = {
+  value?: string;
+  defaultValue?: string;
+  onChange?: (value: string) => void;
+  name?: string;
+  size?: 'small' | 'medium' | 'large';
+};
+
+export const ToggleGroupContext = createContext<ToggleGroupContextProps>({});
+
+export type ToggleGroupProps = {
+  /** Value of the ToggleGroup component. Set when you want to controll the value of the component. */
+  value?: string;
+  /** Default value of the ToggleGroup component. Set when you want to use the component as an uncontrolled component. */
+  defaultValue?: string;
+  /** Function that is called when the value of the ToggleGroup component changes. The current value is available in the callback. */
+  onChange?: (value: string) => void;
+  /** Name of the ToggleGroup component. */
+  name?: string;
+  /** Size of the ToggleGroup component. */
+  size?: 'small' | 'medium' | 'large';
+} & Omit<HTMLAttributes<HTMLDivElement>, 'value' | 'onChange'>;
+
+export const ToggleGroup = forwardRef<HTMLDivElement, ToggleGroupProps>(
+  (
+    { children, value, defaultValue, onChange, size = 'medium', name, ...rest },
+    ref,
+  ) => {
+    const isControlled = value !== undefined;
+    const [uncontrolledValue, setUncontrolledValue] = useState<
+      string | undefined
+    >(defaultValue);
+
+    let onValueChange = onChange;
+    if (!isControlled) {
+      onValueChange = (newValue: string) => {
+        setUncontrolledValue(newValue);
+        onChange?.(newValue);
+      };
+      value = uncontrolledValue;
+    }
+    const nameId = useId();
+
+    return (
+      <div
+        {...rest}
+        className={cn(classes.toggleGroupContainer, rest.className)}
+        ref={ref}
+      >
+        <ToggleGroupContext.Provider
+          value={{
+            value,
+            defaultValue,
+            name: name ?? `radiogroup-name-${nameId}`,
```

Comment:
We should use the name of the component as thats where to data comes from when form is submitted.
```suggestion
            name: name ?? `togglegroup-name-${nameId}`,
```